include/SugarFolders/SugarFolders.php

```diff
@@ -1537,15 +1540,24 @@ protected function shouldFolderDisplay(array $folders, string $folderId): bool
         }
 
         foreach ($folders as $folder) {
+            // Check if unselected in user preferences
             $isSelected = $folder['selected'] ?? false;
             if (isFalse($isSelected)) {
                 continue;
             }
-            $id = $folder['id'] ?? '';
-
-            if ($id === $folderId) {
+            // Check parent ID
+            if ($folder['id'] === $folderId) {
                 return true;
             }
+
+            // Check child IDs
+            if ($folder['has_child']) {
```

Comment:
Hi @MikeyJC, maybe also a good idea to have the `??` operator here, it may avoid notices and warnings.